snappy_pipeline/workflows/cbioportal_export/__init__.py

```diff
@@ -268,77 +406,138 @@ def get_resource_usage(self, action):
         # Validate action
         self._validate_action(action)
         return ResourceUsage(
-            threads=4,
+            threads=2,
             time="12:00:00",  # 12 hours
-            memory="5120M",
+            memory="8192M",
         )
 
 
-class cbioportalMafStepPart(cbioportalExportStepPart):
-    """Helper class to get all required maf files for cbioportal."""
+class cbioportalCnaFilesStepPart(cbioportalExportStepPart):
+    """Generate cbioportal continuous (log2), discrete (pseudo-gistic) & segment tables"""
 
     #: Step name
-    name = "cbioportal_maf"
+    name = "cbioportal_cna"
 
-    @listify
-    def get_input_files(self, action):
-        """Return list of all input files"""
+    #: Extraction type, must be instantiated in sub-class
+    extraction_type = "DNA"
+
+    #: Actions
+    actions = ("log2", "gistic")
+
+    def __init__(self, parent):
+        super().__init__(parent)
+        name_pattern = "bwa." + self.config["copy_number_tool"] + ".{library_name}"
+        self.input_tpl = os.path.join("work/cna", name_pattern, "out", name_pattern + ".cna")
+
+    def get_args(self, action):
+        # Validate action
+        self._validate_action(action)
+        if action == "log2":
+            return {"action_type": "log2", "extra_args": {"pipeline_id": "ENSEMBL"}}
+        if action == "gistic":
+            return {
+                "action_type": "gistic",
+                "extra_args": {"pipeline_id": "ENSEMBL", "amplification": "7"},
```

Comment:
What does the "7" refer to?

Reply:
Should be "9", actually. It sets the minimal value of the copy number to be flagged as an amplification, rather than a gain. This is now made clearer in the code (some comments)